### PR TITLE
Adjusting rescale value for small/large numbers

### PR DESCRIPTION
**Pull request by snmln**
**Related Ticket:** https://github.com/orgs/NASA-IMPACT/projects/17?pane=issue&itemId=85258058&issue=NASA-IMPACT%7Cveda-ui%7C1226

### Description of Changes
- Adjusting the text input to be wider. 
- Cleaning out colorRangeSlider component, creating util file, creating initial tests. 
- Creating dynamic step calculation functionality

### Notes & Questions About Changes
When observing a value in either the max or min inputs if that value has a character count of 6 or greater the component will convert that number into scientific notation as a display value. 

### Validation / Testing
Run this branch on the veda-config code base, navigate to the E&A page and select the difference CO2 dataset. This is an extreme data set that should display all changes in this component. 
- Confirm that the middle range identifier is always at halfway mark. 
- Confirm that the numbers are displaying in scientific notation.

### CODE REVIEW

File: app/scripts/components/exploration/components/datasets/colorRangeSlider/color-range-slider.scss [hunk 1]
```diff
@@ -48,3 +48,24 @@ input::-webkit-inner-spin-button {
 input[type='number'] {
   -moz-appearance: textfield;
 }
+
+.tooltiptext {
+  background-color: #2c3e50;
+}
+
+.tooltipmin::after {
```
Comment (snmln) [inline app/scripts/components/exploration/components/datasets/colorRangeSlider/color-range-slider.scss:56]: @dzole0311 Would you by chance know of a way to leverage pseudo classes in combination with the USWDS util classes?

**Comment by aboydnw** [comment]
@snmln I assume you're looking at NO2 diff, not CO2 diff. Just commenting here for posterity, I think this is an acceptable visualization of these small/large numbers. 

image

Only question is, were you able to test this out where the magnitude of the min/max were both <0.01?

**Comment by snmln** [comment]
CO2 will also display a 1.5e-6 value I believe so both NO2 and CO2 will work for testing purposes. And yes, I was able to confirm the 2 extremes of the data visualization will present as desired.
Screenshot 2024-11-22 at 2 43 42 PM

**Review by aboydnw — APPROVED** [review]
Manual testing looks good on this one. Still need a review from an engineer

File: app/scripts/components/exploration/components/datasets/colorRangeSlider/color-range-slider.scss (hunk 1 above)
Comment (snmln) [reply app/scripts/components/exploration/components/datasets/colorRangeSlider/color-range-slider.scss:56]: We discussed this question async, and determined that USWDS currently doesn't support pseudo class functionality for this usecase.

File: app/scripts/components/common/map/style-generators/raster-timeseries.tsx
Comment (snmln) [inline app/scripts/components/common/map/style-generators/raster-timeseries.tsx:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

File: app/scripts/components/common/map/types.d.ts
Comment (snmln) [inline app/scripts/components/common/map/types.d.ts:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

File: app/scripts/components/common/uswds/index.tsx
Comment (snmln) [inline app/scripts/components/common/uswds/index.tsx:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

File: app/scripts/components/common/uswds/input.tsx
Comment (snmln) [inline app/scripts/components/common/uswds/input.tsx:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

File: app/scripts/components/exploration/atoms/hooks.ts
Comment (snmln) [inline app/scripts/components/exploration/atoms/hooks.ts:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

File: app/scripts/components/exploration/components/datasets/colormap-options.tsx
Comment (snmln) [inline app/scripts/components/exploration/components/datasets/colormap-options.tsx:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

File: app/scripts/components/exploration/components/datasets/data-layer-card.tsx
Comment (snmln) [inline app/scripts/components/exploration/components/datasets/data-layer-card.tsx:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

File: app/scripts/components/exploration/components/datasets/dataset-list-item.tsx
Comment (snmln) [inline app/scripts/components/exploration/components/datasets/dataset-list-item.tsx:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

File: app/scripts/components/exploration/components/map/layer.tsx
Comment (snmln) [inline app/scripts/components/exploration/components/map/layer.tsx:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

File: app/scripts/components/exploration/types.d.ts.ts
Comment (snmln) [inline app/scripts/components/exploration/types.d.ts.ts:1]: This work was previously approved in: https://github.com/NASA-IMPACT/veda-ui/pull/1190. Then reverted when bug was identified. I needed to pull those changes back in to correctly reflect updates.

**Comment by snmln** [comment]
@vgeorge Thank you for these call out! I wanted to respond to a few: 

> When entering values outside the range, the error message displays them on a different format:
This is purposeful. The scientific notation is a solution to ensure as much of the number is visible within the boundaries of the input space as possible. 


> Rescale changes are no applied after changing the slides, I needed to switch color scales to see the changes reflected on the map:

Good catch there was a few raster-layer changes that got lost in the shuffle with re-implementing the changes from https://github.com/NASA-IMPACT/veda-ui/pull/1190. I have gone ahead an implemented them and it should now be functioning correctly.

**Pull request merged**